### PR TITLE
infra: validate git status after commands

### PR DESCRIPTION
**Pull request by romani**
from https://github.com/checkstyle/checkstyle/pull/11806

update to catch unstaged files

### CODE REVIEW

**Comment by romani** [comment]
it works:
https://cloud.drone.io/checkstyle/checkstyle/11062/3/4

```
[INFO] BUILD SUCCESS

[INFO] ------------------------------------------------------------------------

[INFO] Total time:  44.002 s

[INFO] Finished at: 2022-07-03T00:01:12Z

[INFO] ------------------------------------------------------------------------

version:10.4-SNAPSHOT

+ ./.ci/validation.sh git-diff

Please clean up or update .gitattributes file.

Git status output:

Top 300 lines of diff:

On branch master

Your branch is ahead of 'origin/master' by 2 commits.

  (use "git push" to publish your local commits)


Untracked files:

  (use "git add <file>..." to include in what will be committed)

	dependency-reduced-pom.xml


nothing added to commit but untracked files present (use "git add" to track)
```

File: .ci/validation.sh [hunk 1]
```diff
@@ -591,7 +590,7 @@ no-error-pgjdbc)
   checkout_from https://github.com/pgjdbc/pgjdbc.git
   cd .ci-temp/pgjdbc
   # pgjdbc easily damage build, we should use stable versions
-  git checkout "261181f31c0eb""e2deb593c1cc51174898ad6c50c"
+  git checkout "417c9a2354ad""c3d2c80f84b0a5059ce""ad92e7c2b"
```
Comment (rnveach) [inline .ci/validation.sh:593]: Why?

Comment (romani) [reply .ci/validation.sh:593]: Ups, I am not sure how it happened, I will revert such changes

Comment (romani) [reply .ci/validation.sh:593]: Pushed

**Pull request merged**